### DISCUSSION TITLE
Another build

### DISCUSSION THREAD

**Discussion by rufusms** [micbay/RaceTimerAndController#24] [Show and tell]
Thank you for sharing the best lap counter and chronometer that I have seen, recently I decided to restore my Scalextric track and I also decided to make a setup that would help me do races and time the laps and I came across your circuit, I thought it was extraordinary In addition to the fact that it is very commented and documented, I thought it was so good that I started to build it although I have to apologize because I have introduced some changes in my assembly, the first has to do with the arduino, it is an atmega 328 stanalone, so it has a few less inputs, although almost everything worked, (from the beginning the only thing I haven't been able to get working is the traffic light), the second change is the keyboard, I added a pcf8574 and connected it through i2c, I just had to delete the music so I could compile it, it lacks memory, I have made a design of the pillar with fusion360 and I think it is looking good, I hope you like it
![IMG_1592](https://github.com/user-attachments/assets/8395b390-ef54-414e-b495-3b9193c69cbe)
![IMG_1589](https://github.com/user-attachments/assets/b1f635b9-0f00-41b1-aeea-7c4b1a6cdc20)
![IMG_1588](https://github.com/user-attachments/assets/780cbe6c-b337-4dc2-9622-b3709e060c95)
![IMG_1587](https://github.com/user-attachments/assets/92c451c6-3e2e-46e6-958e-d9f831c3e874)
![IMG_1586](https://github.com/user-attachments/assets/78d9bec3-d2b5-485c-9ff4-1dd0069fee64)
![IMG_1585](https://github.com/user-attachments/assets/6f9a0db4-6e97-42b9-9160-d4d6038f9872)
![IMG_1584](https://github.com/user-attachments/assets/03960dcd-3b18-42b0-b77a-5c36ecc76c0c)
![IMG_1582](https://github.com/user-attachments/assets/f7a4ca04-2afe-4124-a817-ef711dadb286)
![IMG_1583](https://github.com/user-attachments/assets/c704e0a1-35da-4189-abd2-6b8ff122cf22)
![IMG_1581](https://github.com/user-attachments/assets/3da60f33-b259-4ae8-a53d-888a34b1ca95)
![IMG_1580](https://github.com/user-attachments/assets/4ee5a38b-0c12-44b2-b624-4158962fa0c5)
![IMG_1579](https://github.com/user-attachments/assets/5c24b026-8183-49b5-af45-91d02016cefc)

**Comment by micbay** [comment]
Wow! Amazing build, it looks great, thanks so much for sharing.

**Comment by micbay** [comment]
Oh, i forgot to mention. Perhaps you already did this and still needed more memory and had to remove the song data, but if you haven't, you aren't using the Adafruit start light, so you could gain a good bit of memory by removing the Adafruit GFX, and Adafruit LED Backpack libraries. Doing this you'll need to also search for the few places it's used.
- Comment out Adafruit code from line 47-73
    - Search for usage of variable `clearStartLight` and comment those lines out, or remove from any conditional statements used. NOTE: some places might be using this boolean flag with other variables, so don't just comment out the whole line of code if it is being used this way, just remove the `clearStartLight` variable from the expression.
    - Search for calls to the function `setBargraph()` and comment them out.

**Reply by rufusms** [reply]
Thanks Micbay, I have deleted all the lines that refer to the ledbar, so now there are approximately 6k available and 8 free lines on the arduino, I would like to do something more with this, but at the moment I can't think of anything, I'm not good at programming so it will probably take me a long time to do something, in any case this will make everything go more smoothly, thank you
